### PR TITLE
Extend BatchNorm with rank-matched channels and optional momentum

### PR DESCRIPTION
**Pull request by rsuderman**
- **Rank-matched channel tensors**: SCALE, BIAS, MEAN, VAR, SAVED_MEAN, and SAVED_INV_VARIANCE now accept tensors of shape [1, C, 1, ..., 1] (ones in all non-feature dimensions) in addition to the canonical 1D [C] shape. The node validates and infers strides for both forms.  The ASM emitter collapses rank-matched inputs to 1D via `flatten.using_ints` before `native_batch_norm`, and expands rank-matched stat outputs back via `reshape` after.

- **Optional momentum**: Momentum is no longer required.  When omitted, the ASM emitter emits `torch.constant.float 1.000000e-01` (PyTorch default). If provided it must still be a scalar constant.

- **Tests**:
  - `test_batchnorm_node.cpp`: updated "Momentum missing" section to expect success; added rank-matched validation sections.
  - `test_batchnorm_infer_asm_emitter_nchw_rank_matched.cpp`: lit test verifying `flatten.using_ints` collapse for rank-matched MEAN/VAR.
  - `test_batchnorm_infer_asm_emitter_nchw_no_momentum.cpp`: lit test verifying `torch.constant.float` default when momentum is absent.

- **Samples**:
  - `batchnorm_infer_nchw_rank_matched_scale_bias.cpp`: end-to-end inference with [1,C,1,1] scale, bias, mean, var.
  - `batchnorm_infer_nchw_no_momentum.cpp`: end-to-end inference with no momentum tensor.
  - `batchnorm_infer_ncdhw_rank_matched.cpp`: end-to-end 5D (NCDHW) inference with rank-matched [1,C,1,1,1] channel tensors.

### CODE REVIEW

File: include/fusilli/node/batchnorm_node.h [hunk 1]
```diff
@@ -125,12 +125,12 @@ class BatchNormNode : public NodeCRTP<BatchNormNode> {
     FUSILLI_RETURN_ERROR_IF(!eT->isScalar(), ErrorCode::InvalidAttribute,
                             "BatchNorm epsilon must be a scalar constant");
 
-    // Momentum checks.
+    // Momentum checks (optional — omitting uses the PyTorch default 0.1).
```
[the remark this answers is not in the document]
Comment (IanWood1) [reply include/fusilli/node/batchnorm_node.h]: >However pytorch does not even use it for training mode. Its just kinda buggy.

I'm not sure what this means. Does pytorch always omit momentum in their testing and just use the defafult (0.1)?
